msal/tests/Test.MSAL.NET.Unit.net45/CoreTests/InstanceTests/B2cAuthorityTests.cs

```diff
@@ -80,29 +85,40 @@ public void NotEnoughPathSegmentsTest()
         [TestCategory("B2CAuthorityTests")]
         public void ValidationEnabledNotSupportedTest()
         {
-            var instance = Authority.CreateAuthority(CoreTestConstants.B2CAuthority, true);
-            Assert.IsNotNull(instance);
-            Assert.AreEqual(instance.AuthorityType, AuthorityType.B2C);
-            try
+            using (var httpManager = new MockHttpManager())
             {
-                Task.Run(
-                    async () =>
+                //add mock response for instance validation
+                httpManager.AddMockHandler(
+                    new MockHttpMessageHandler
                     {
-                        await instance.ResolveEndpointsAsync(
-                            null, 
-                            null,
-                            null, 
-                            new RequestContext(null, new TestLogger(Guid.NewGuid(), null))).ConfigureAwait(false);
-                    }).GetAwaiter().GetResult();
-                Assert.Fail("test should have failed");
-            }
-            catch (Exception exc)
-            {
-                Assert.IsInstanceOfType(exc, typeof(ArgumentException));
-                Assert.AreEqual(CoreErrorMessages.UnsupportedAuthorityValidation, exc.Message);
+                        Method = HttpMethod.Get,
+                        Url = "https://login.microsoftonline.com/tfp/tenant/policy/v2.0/.well-known/openid-configuration",
+                        ResponseMessage =
+                            MockHelpers.CreateSuccessResponseMessage(
+                                File.ReadAllText(ResourceHelper.GetTestResourceRelativePath("OpenidConfiguration-B2C.json")))
+                    });
+
+                var instance = Authority.CreateAuthority(CoreTestConstants.B2CAuthority, true);
+                Assert.IsNotNull(instance);
+                Assert.AreEqual(instance.AuthorityType, AuthorityType.B2C);
+                try
+                {
+                    Task.Run(
+                        async () =>
+                        {
+                            await instance.ResolveEndpointsAsync(
+                                httpManager,
+                                new TelemetryManager(),
+                                null,
+                                new RequestContext(null, new TestLogger(Guid.NewGuid(), null))).ConfigureAwait(false);
+                        }).GetAwaiter().GetResult();
+                }
```

Comment:
>} [](start = 16, length = 1)

seems like you are expecting an exception. 
1: Please only catch the exception type you are expecting.
2: Please ensure to Assert.Fail, for the "happy" scenario as that is not expected. (the code you cleaned-up had this)